clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3331,7 +3338,19 @@ private PubSubMessageProcessedResult processMessage(
         if (isWriteComputationEnabled && partitionConsumptionState.isEndOfPushReceived()) {
           partitionConsumptionState.setTransientRecord(kafkaClusterId, consumerRecord.getOffset(), keyBytes, -1, null);
         }
-        return new PubSubMessageProcessedResult(new WriteComputeResultWrapper(null, null, false));
+        // Best-effort to provide the old value for delete operation in case needed by a ComplexVeniceWriter to generate
```

Comment:
This is actually pretty interesting.  There.... would be many edge cases where this wouldn't work out...  But....  I don't think we could do any better lol.

Really, this should be a great big warning label that view logic needs to be idempotent.  Otherwise all bets are off.

Reply:
I agree... but like Gaojie mentioned in earlier comments, it could be useful in the event where we could propagate the delete properly to view topics even when a value based partitioner is used.